services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/multipart/KnownContentLengthAsyncRequestBodySubscriber.java

```diff
@@ -201,7 +201,7 @@ private void completeMultipartUploadIfFinished(int requestsInFlight) {
     }
 
     private CompletedPart[] mergeCompletedParts() {
-        CompletedPart[] merged = new CompletedPart[partCount];
+        CompletedPart[] merged = new CompletedPart[(int) partCount];
```

Comment:
This seems dangerous. Looks like max part number is 10,000 and I don't think it'll ever exceed max integer value... Should we change the type to integer then?

Reply:
updated `partCount` and `numExistingParts` to int